### PR TITLE
[feature/azure_data_explorer]

### PR DESCRIPTION
**Pull request by ag-ramachandran**
**Description:** 

Adding a new exporter to OpenTelemetry that supports Metrics, Logs and Traces into [Azure Data Explorer](https://azure.microsoft.com/en-us/services/data-explorer/), a timeseries & analytics offering from Azure

**Link to tracking Issue:** 

[#11830 Issue for new component that was approved](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11830)

**Testing:** 

a) Unit tests to cover 88% of code
b) Functional end to end tests with actual provisioned database
c) Ingest volumes by enabling JAVA agent and running a load test on API's to test load

**Documentation:** 

Elaborate documentation is provided in the README section of the request, that contains elaborate set up and configuration needed for the exporter. Post approval, complete documentation will also be provided on the Microsoft documentation page as well

### CODE REVIEW

**Review by codeboten — COMMENTED** [review]
Thanks for the PR, I've done an initial pass and added some comments. Will do a second review once these are resolved!

**Comment by asaharn** [comment]
Hi @codeboten , the comments have been resolved. Requesting you for a re-review. Thank you.

**Comment by Kiran-G1** [comment]
Amazing work sensei !!! @ag-ramachandran Thanks for this.

**Comment by ag-ramachandran** [comment]
@codeboten when you're back and get some time, will be great to take any next steps. Pl. let us know if we can help with something to get this merged

File: exporter/azuredataexplorerexporter/README.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+# Azure Data Explorer Exporter
+
+| Status                   |           |
+| ------------------------ | --------- |
+| Stability                | [alpha]   |
```
Comment (codeboten) [inline exporter/azuredataexplorerexporter/README.md:5]: You'll need to add the links to `alpha` and `contrib`, see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d444f478e7478fdc31c46533136a9ba6bd9f433e/exporter/coralogixexporter/README.md?plain=1#L57-L59

File: unreleased/bugfix_OTELCoreReviewComments.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+change_type:'new_component'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component:'azuredataexplorer'
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note:"Adding a new exporter to OpenTelemetry that supports Metrics, Logs and Traces into Azure Data Explorer, a timeseries & analytics offering from Azure"
+
+# One or more tracking issues related to the change
+issues: [https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11830]
```
Comment (codeboten) [inline unreleased/bugfix_OTELCoreReviewComments.yaml:11]:
```suggestion
issues: [11830]
```

File: exporter/azuredataexplorerexporter/tracesdata_to_adx.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type AdxTrace struct {
+	TraceID            string                 // TraceId associated to the Trace
+	SpanID             string                 // SpanId associated to the Trace
+	ParentID           string                 // ParentId associated to the Trace
```
Comment (codeboten) [inline exporter/azuredataexplorerexporter/tracesdata_to_adx.go:26]:
```suggestion
	TraceID            string                 // TraceID associated to the Trace
	SpanID             string                 // SpanID associated to the Trace
	ParentID           string                 // ParentID associated to the Trace
```

File: exporter/azuredataexplorerexporter/tracesdata_to_adx_test.go [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	tmap := make(map[string]interface{})
+	tmap["key"] = "value"
+	tmap[hostkey] = testhost
+
+	spanID := [8]byte{0, 0, 0, 0, 0, 0, 0, 50}
+	traceID := [16]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 100}
+
+	tests := []struct {
+		name             string             // name of the test
+		spanDatafn       func() ptrace.Span // function that generates the
```
Comment (codeboten) [inline exporter/azuredataexplorerexporter/tracesdata_to_adx_test.go:38]: Looks like the comment is missing something here.

File: exporter/azuredataexplorerexporter/config.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	if adxCfg == nil {
+		return errors.New("ADX config is nil / not provided")
+	}
+
+	if isEmpty(adxCfg.ClusterURI) || isEmpty(adxCfg.ApplicationID) || isEmpty(adxCfg.ApplicationKey) || isEmpty(adxCfg.TenantID) {
+		return errors.New(`mandatory configurations "cluster_uri" ,"application_id" , "application_key" and "tenant_id" are missing or empty `)
+	}
+
+	if !(adxCfg.IngestionType == managedIngestType || adxCfg.IngestionType == queuedIngestTest || isEmpty(adxCfg.IngestionType)) {
+		return fmt.Errorf("unsupported configuration for ingestion_type. Accepted types [%s, %s] Provided [%s]", managedIngestType, queuedIngestTest, adxCfg.IngestionType)
```
Comment (codeboten) [inline exporter/azuredataexplorerexporter/config.go:53]: Please add a test for this condition

**Review by codeboten — COMMENTED** [review]
Overall this looks good, just a few comments and this will likely need rebasing after this week's release (which should be completed within the next day)

File: exporter/azuredataexplorerexporter/config.go (hunk 5 above)
Comment (ag-ramachandran) [reply exporter/azuredataexplorerexporter/config.go:53]: added a test case that covers this scenario

**Review by codeboten — APPROVED** [review]
Please resolve conflicts and we can get this merged

**Comment by ag-ramachandran** [comment]
Thanks @codeboten , merged and resolved all conflicts

**Review by codeboten — APPROVED** [review]

**Pull request merged**